aiida/orm/querybuilder.py

```diff
@@ -1766,6 +1766,23 @@ def get_json_compatible_queryhelp(self):
             'offset': self._offset,
         })
 
+    @property
+    def queryhelp(self):
+        """queryhelp dictionary correspondig to QueryBuilder instance.
+
+        The queryhelp can be used to create a copy of the QueryBuilder instance like so::
+
+            qb = QueryBuilder(limit=3).append(StructureData, project='id').order_by({StructureData:'id'})
+            qb2=QueryBuilder(**qb.queryhelp)
+
+        :return: a queryhelp dictionary
+        """
+        return self.get_json_compatible_queryhelp()
+
+    def __deepcopy__(self, memodict={}):
```

Comment:
Why only `__deepcopy__` implementation. Wouldn't a normal copy also not work? Since the queryhelp won't contain any references to other objects a shallow copy would also just work when implemented like this no?

Reply:
Well, I chose this because the implementation of the queryhelp uses `deepcopy`.
https://github.com/aiidateam/aiida-core/blob/5a1f18b67599ce25766cffea9a00c7e420141f6d/aiida/orm/querybuilder.py#L1760-L1767

And the deepcopy is actually necessary since it contains nested dictionaries:
```
In [1]: qb=QueryBuilder()

In [2]: qb.append(Data)
Out[2]: <aiida.orm.querybuilder.QueryBuilder at 0x11ea02208>

In [3]: qb.queryhelp
Out[3]:
{'path': [{'entity_type': 'data.Data.',
   'tag': 'Data_1',
   'joining_keyword': None,
   'joining_value': None,
   'outerjoin': False,
   'edge_tag': None}],
 'filters': {'Data_1': {'node_type': {'like': 'data.%'}}},
 'project': {'Data_1': []},
 'order_by': {},
 'limit': None,
 'offset': None}
```